src/main/java/com/breskul/bibernate/persistence/GenericDao.java

```diff
@@ -84,4 +91,49 @@ private <T> T mapResult(ResultSet resultSet, Class<T> cls) {
           .formatted(cls), e);
     }
   }
+
+  public <T> int executeUpdate(EntityKey<T> entityKey, Object... parameters) {
+    String updateSql = prepareUpdateQuery(entityKey);
+    log.trace("Update entity: [{}]", updateSql);
+    try (Connection connection = getConnection();
+        PreparedStatement preparedStatement = connection.prepareStatement(updateSql)) {
+      setParameters(preparedStatement, entityKey.id(), parameters);
+      return preparedStatement.executeUpdate();
+    } catch (SQLException e) {
+      throw new BibernateException("Failed to execute update query: [%s] with parameters %s"
+          .formatted(updateSql, Arrays.toString(parameters)), e);
+    }
+  }
+
+  public <T> String prepareUpdateQuery(EntityKey<T> entityKey) {
+    Class<T> entityClass = entityKey.entityClass();
+    String setUpdatedColumnsSql = EntityUtil.getEntityColumnNames(entityClass).stream()
+        .map("%s = ?"::formatted)
+        .collect(Collectors.joining(", "));
+    String tableName = EntityUtil.getEntityTableName(entityClass);
+    String primaryKeyName = EntityUtil.findEntityIdFieldName(entityClass);
+
+    return UPDATE_SQL.formatted(tableName, "SET %s".formatted(setUpdatedColumnsSql),
+        primaryKeyName);
+  }
+
+  public <T> void setParameters(PreparedStatement preparedStatement, Object pk,
+      Object... parameters) throws SQLException {
+    int i = 1;
+    for (Object parameter : parameters) {
+      preparedStatement.setObject(i++, parameter);
+    }
+
+    if (pk != null) {
```

Comment:
Should we throw exception if Id is null?

Reply:
done